gcloud_bigtable/google/bigtable/v1/bigtable_service_api.py

```diff
@@ -0,0 +1,175 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# EDITING INSTRUCTIONS
+# This file was generated from google/bigtable/v1/bigtable_service.proto,
+# and updates to that file get reflected here through a regular refresh
+# process. However, manual and updates to that file get reflected here
+# through a regular refresh additions are allowed because the refresh
+# process performs a 3-way merge in order to preserve those manual additions.
+# In order not to break the refresh process, only certain types of
+# modifications are allowed.
+#
+# Allowed modifications:
+# 1. New methods (these should be added to the end of the class)
+# 2. "Notes specific to this wrapper method" sections in the method
+#    documentation
+#
+# Happy editing!
+
+from google.bigtable.v1 import bigtable_data_pb2
+from google.bigtable.v1 import bigtable_service_messages_pb2
+from google.bigtable.v1 import bigtable_service_pb2
+from google.gax import api_callable
+from google.gax import api_utils
+from google.gax import page_descriptor
+
+class BigtableServiceApi(object):
+  """Service for reading from and writing to existing Bigtables."""
+
+  # The default address of the logging service.
+  _SERVICE_ADDRESS = "bigtable.googleapis.com"
+
+  # The default port of the logging service.
+  _DEFAULT_SERVICE_PORT = 443
+
+  # The scopes needed to make gRPC calls to all of the methods defined in
+  # this service
+  _ALL_SCOPES = [
+      'https://www.googleapis.com/auth/bigtable.data',
+      'https://www.googleapis.com/auth/cloud-bigtable.data',
+      'https://www.googleapis.com/auth/cloud-platform'
+  ]
+
+  def __init__(
+      self, service_path=_SERVICE_ADDRESS, port=_DEFAULT_SERVICE_PORT,
+      channel=None, ssl_creds=None, scopes=_ALL_SCOPES,
+      is_idempotent_retrying=True, max_attempts=3, timeout=30):
+
+    self.defaults = api_callable.ApiCallableDefaults(
+      timeout=timeout, max_attempts=max_attempts,
+      is_idempotent_retrying=is_idempotent_retrying)
+
+    self.stub = api_utils.create_stub(
+      bigtable_service_pb2.beta_create_BigtableService_stub, service_path, port,
+      ssl_creds=ssl_creds, channel=channel, scopes=scopes)
+
+  def __enter__(self):
+    return self
+
+  def __exit__(self, type, value, traceback):
+    self.close()
+
+  def close(self):
+    del self.stub
+
+  # Properties
+  @property
+  def channel(self):
+    return self.channel
+
+  # Page descriptors
+
+  # Service calls
+  def read_rows(self, table_name="", row_key="", row_range=bigtable_data_pb2.RowRange(), filter=bigtable_data_pb2.RowFilter(), allow_row_interleaving=False, num_rows_limit=0L, **kwargs):
```

Comment:
It doesn't seem like much is different about this method.

However, `read_rows` returns a `stream` so it should be very different. (It is `__iter__`-able in the gRPC generated code.)